src/pybind/mgr/prometheus/module.py

```diff
@@ -53,7 +53,8 @@ def os_exit_noop(status: int) -> None:
 
 _global_instance = None  # type: Optional[Module]
 cherrypy.config.update({
-    'response.headers.server': 'Ceph-Prometheus'
+    'response.headers.server': 'Ceph-Prometheus',
+    'tools.gzip.on': True,
```

Comment:
@epuertat  Maybe adding `cache-control` too? Not sure if it'll impact performance much or not.